homeassistant/components/arcam_fmj/device_trigger.py

```diff
@@ -0,0 +1,70 @@
+"""Provides device automations for Arcam FMJ Receiver control."""
+from typing import List
+
+import voluptuous as vol
+
+from homeassistant.components.automation import AutomationActionType
+from homeassistant.components.device_automation import TRIGGER_BASE_SCHEMA
+from homeassistant.const import (
+    CONF_DEVICE_ID,
+    CONF_DOMAIN,
+    CONF_PLATFORM,
+    CONF_TYPE,
+    CONF_ZONE,
+)
+from homeassistant.core import CALLBACK_TYPE, Event, HomeAssistant, callback
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DEVICE_ZONES, DOMAIN, SIGNAL_TURN_ON
+
+CONF_SUBTYPE = "subtype"
+
+TRIGGER_TYPES = {"turn_on"}
+TRIGGER_SUBTYPES = [str(x) for x in DEVICE_ZONES]
+TRIGGER_SCHEMA = TRIGGER_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_SUBTYPE): vol.In(TRIGGER_SUBTYPES),
+        vol.Required(CONF_TYPE): vol.In(TRIGGER_TYPES),
+    }
+)
+
+
+async def async_get_triggers(hass: HomeAssistant, device_id: str) -> List[dict]:
+    """List device triggers for Arcam FMJ Receiver control devices."""
+    triggers = []
+
+    for subtype in TRIGGER_SUBTYPES:
+        triggers.append(
+            {
+                CONF_PLATFORM: "device",
+                CONF_DEVICE_ID: device_id,
+                CONF_DOMAIN: DOMAIN,
+                CONF_SUBTYPE: subtype,
+                CONF_TYPE: "turn_on",
+            }
+        )
+
+    return triggers
+
+
+async def async_attach_trigger(
+    hass: HomeAssistant,
+    config: ConfigType,
+    action: AutomationActionType,
+    automation_info: dict,
+) -> CALLBACK_TYPE:
+    """Attach a trigger."""
+    config = TRIGGER_SCHEMA(config)
+
+    if config[CONF_TYPE] == "turn_on":
+
+        @callback
+        def _handle_event(event: Event):
+            if event[CONF_ZONE] == config[CONF_SUBTYPE]:
+                hass.async_run_job(action({"trigger": config}))
+
+        return hass.helpers.dispatcher.async_dispatcher_connect(
+            SIGNAL_TURN_ON, _handle_event
+        )
```

Comment:
I have mixed feelings about this idea. 

1. Device Automations have been limited to use the automation building blocks. This way we make sure that we don't force device automations on people that don't want to. (one day I even want to allow "conversions")
2. I think it's a great idea because once we implement the idea from the mockup that we suggest to create an automation, it will suggest probably with this one, making it clear from the device info page that it is available.
  ![image](https://user-images.githubusercontent.com/1444314/81619046-8f353980-939d-11ea-9734-87b7e3e56d3d.png)
3. It is a hack. If we ever end up implementing device options, this could be just one of the options. But who knows when that will happen 🤷 and we can convert then.

Reply:
1 can be solved by making it an event btw.

Reply:
We normally don't want entities to fire custom events. Should the event in that case be fired separately from the entity by a device api representing instance?

Reply:
Should we make a "fake" state change in the entity when turning on, which will spawn an event?

Reply:
I think in this case we should make an exception and just fire an event.

Reply:
What will our guideline be then? When is it ok to fire events from the entity? When is it not ok?

Reply:
Only if we need to allow plugging in a turn_on command ?

Reply:
Ok.

Reply:
So back to events now.